service/history/workflow/update/registry.go

```diff
@@ -163,16 +164,30 @@ func NewRegistry(
 	}
 
 	getStoreFn().VisitUpdates(func(updID string, updInfo *updatespb.UpdateInfo) {
-		// need to eager load here so that Len and admit are correct.
-		if acc := updInfo.GetAcceptance(); acc != nil {
+		if updInfo.GetRequest() != nil {
```

Comment:
When will the update registry be notified about the reapplied update so that update response from sdk won't be rejected by server?

Reply:
I think for reset it works because it's a new workflow, so following requests on this workflow will create a new workflow context and build a new registry. 

For conflict resolution case I don't think ^ applies? Actually even without conflict resolution. During normal replication & failover case, how does update registry know about replicated requested/accepted updates?

Reply:
> I think for reset it works because it's a new workflow, so following requests on this workflow will create a new workflow context and build a new registry.

Agreed

> For conflict resolution case I don't think ^ applies? Actually even without conflict resolution. During normal replication & failover case, how does update registry know about replicated requested/accepted updates?

How about we merge this PR now, and study this question in https://github.com/temporalio/temporal/pull/5595 which adds helpful test machinery? In #5595 I've added a test named `TestAcceptedUpdateCanBeCompletedAfterFailover` that demonstrates that an accepted update can be completed after failover, but it's possible that there are cases that need more work, or that there is something wrong in the way that that test is reproducing the scenario.